### PR TITLE
chore: add snasphot and rebuild history to bundle, disable resouce level dump

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
- Add VolumeSnapshot and VolumeSnapshotContent CRs for mayastor driver.
- Add Rest api snapshot resources to topology.
- Add Rebuild history to the volume.
- Remove dead code and crate.
- Hide resource level dump commands, until further refactor.

### CODE REVIEW

File: k8s/plugin/README.md [hunk 1]
```diff
@@ -316,58 +310,23 @@ Supportability - collects state & log information of services and dumps it to a
...
-      kubectl mayastor dump system -d /mayastor-dump -n mayastor
-      ```
-    - Example command while running outside of Kubernetes cluster nodes where
-      nodes exist in private network (or) node ports are not exposed for outside cluster
-      ```sh
-      kubectl mayastor dump system -d /mayastor-dump -r http://127.0.0.1:30011 -l http://127.0.0.1:3100 -e http://127.0.0.1:2379 -n mayastor
-      ```
-
-2. To collect information about all mayastor volumes into an archive file
+ - Example command while running inside Kubernetes cluster nodes / system which
```
Comment (tiagolobocastro) [inline k8s/plugin/README.md:319]: We go via kube-api not cluster ports now?

File: k8s/supportability/src/collect/k8s_resources/k8s_resource_dump.rs [hunk 2]
```diff
@@ -212,6 +214,46 @@ impl K8sResourceDumperClient {
             }
         }
 
+        log("\t Collecting Kubernetes VolumeSnapshotClass resources".to_string());
```
Comment (tiagolobocastro) [inline k8s/supportability/src/collect/k8s_resources/k8s_resource_dump.rs:217]: Can we put these into a function?

File: k8s/supportability/src/operations.rs [hunk 3]
```diff
@@ -15,36 +15,43 @@ pub(crate) enum Operations {
...
 /// Resources on which operation can be performed
 #[derive(clap::Subcommand, Clone, Debug)]
 pub(crate) enum Resource {
     /// Collects entire system information
-    System,
+    System(SystemDumpArgs),
 
     /// Collects information about all volumes and its descendants (replicas/pools/nodes)
-    #[clap(name = "volumes")]
+    #[clap(name = "volumes", hide = true)]
```
Comment (tiagolobocastro) [inline k8s/supportability/src/operations.rs:32]: Thought these were doing to be debug only rather than always hidden?

**Comment by Abhinandan-Purkait** [comment]
bors try

**Review by sinhaashish — APPROVED** [review]

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**